Scripts/devEnv.js

```diff
@@ -1,162 +1,240 @@
-                BUFFER.push(`[INFO][${file}] extracting stylesheet regexp`);
-                
-                content = replaceSource(content,match,CSS_MAPER,CSS_SOURCES);
-
-            } else {
-                BUFFER.push(`[INFO] ${file} does not contain STYLES directive`);
-            }
-
-            // writing the new file.
-            filename = path.join(DEV_DIR ,path.relative(process.argv[2],file));
-
-            // ensuring that the directory exists.
-            await fs.promises.mkdir(path.dirname(filename),{ recursive: true });
-
-            await fs.promises.writeFile(
-                filename,
-                content
-            );
-        }
-
-        // copying all the files to the build folder
-        let pending = [];
-        
-        JS_SOURCES.forEach(source => {
-            pending.push(
-                fs.promises.copyFile(
-                    path.join(process.argv[3],source),
-                    path.join(DEV_DIR,source)
-                )
-            );
-        });
-
-        CSS_SOURCES.forEach(source => {
-            pending.push(
-                fs.promises.copyFile(
-                    path.join(process.argv[4],source),
-                    path.join(DEV_DIR,source)
-                )
-            );
-        });
-
-        await Promise.all(pending);
-
-    } catch(e) {
-        console.error(e);
-        LOG.write(BUFFER.join("\n"));
-        LOG.write("\nUnexpected while producing dev env\n ");
-        LOG.write(e.stack);
-        process.exit(1);
-    }
-
-    LOG.write(BUFFER.join("\n"));
-    let last = "\nSuccessfully builded dev env in " + (Date.now() - START) + "ms\n";
-    console.log(last);
-    LOG.write(last);
-
+/**
+ * @author Galyfray
+ *
+ * @file this file contains the functions that will be used to create the dev environment.
+ *
+ * @param {String} html the root folder of the html files to be parsed.
+ * @param {String} js the root folder of the js files to be included in the html.
+ * @param {String} css the root folder of the css files to be included in the html.
+ * @param {String} as the root folder of the assembly scripts files to be compiled.
+ */
+
+const {ESLint} = require("eslint");
+const fs = require("fs");
+const path = require("path");
+
+
+/** This function will find every file in the given directory and his subdirectories and return an array of all the file names.
+ *
+ * @param {String} dir the name of the directory to iterate through.
+ */
+async function getAllFiles(dir) {
+    return fs.promises.readdir(dir).then(async files => {
+        const allFiles = [];
+
+        // For each element we check if it is a directory or a file.
+        // If it is a file then we add it to the array and return nothing, the promise.all will stop at the enclosing .then.
+        // If it is a directory we call the function again with the new directory
+        // We then return the new promse which once resolved will add all files to the array.

```

Comment:
new promse → new promise

Reply:
The error is still here

Reply:
The error will remain in this conversation for it to keep sense, however as you can see in [this](https://github.com/galyfray/MO54/pull/3/commits/fe29b02fd951e9ccd524d9a72a4420426ed533f3) commit I fixed it